File: Algorithm/C++/1000+/1290. Convert Binary Number in a Linked List to Integer.cpp
class Solution {
public:
    int getDecimalValue(ListNode* head) {
        int sum = 0;
        while(head != NULL){
            sum += head->val;
            if(head->next == NULL){
                break;
            }else{
                sum *= 2;
                head = head->next;
            }
        }
        return sum;
    }
};

File: Algorithm/Python/1000+/1290. Convert Binary Number in a Linked List to Integer.py
class Solution:
    def getDecimalValue(self, head: ListNode) -> int:
        result = 0
        while head != None:
            result += head.val
            if head.next == None:
                break
            else:
                result *= 2
                head = head.next
        return result

# better
class Solution:
    def getDecimalValue(self, head: ListNode) -> int:
        num = head.val
        while head.next:
            num = num * 2 + head.next.val
            head = head.next
        return num
